components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/internal/Collection+.kt

```diff
@@ -4,25 +4,39 @@
 
 package org.mozilla.experiments.nimbus.internal
 
-public fun <K, V, K1> Map<K, V>.mapKeys(transform: (K) -> K1?): Map<K1, V> =
+fun <K, V, K1> Map<K, V>.mapKeys(transform: (K) -> K1?): Map<K1, V> =
     this.entries
         .mapNotNull { e ->
             transform(e.key)?.let { it to e.value }
         }
         .toMap()
 
-public fun <K, V, V1> Map<K, V>.mapValues(transform: (V) -> V1?): Map<K, V1> =
+fun <K, V, V1> Map<K, V>.mapValues(transform: (V) -> V1?): Map<K, V1> =
     this.entries
         .mapNotNull { e ->
             transform(e.value)?.let { e.key to it }
         }
         .toMap()
 
-public fun <K, V, K1, V1> Map<K, V>.mapEntries(keyTransform: (K) -> K1?, valueTransform: (V) -> V1?) =
+fun <K, V, K1, V1> Map<K, V>.mapEntries(keyTransform: (K) -> K1?, valueTransform: (V) -> V1?): Map<K1, V1> =
     this.entries
         .mapNotNull { e ->
             val k1 = keyTransform(e.key) ?: return@mapNotNull null
             val v1 = valueTransform(e.value) ?: return@mapNotNull null
             k1 to v1
         }
         .toMap()
+
+fun <K, V> Map<K, V>.mergeWith(defaults: Map<K, V>, valueTransform: ((V, V) -> V?)? = null) =
+    valueTransform?.let {
+        val target = mutableMapOf<K, V>()
+        defaults.entries.forEach { entry ->
+            target[entry.key] = entry.value
+        }
+        entries.forEach { entry ->
+            val override = defaults[entry.key]
+                ?.let { d -> valueTransform(entry.value, d) }
+            target[entry.key] = override ?: entry.value
+        }
+        target
+    } ?: defaults + this
```

Comment:
This method simplifies the generated code massively.

Reply:
Nice!